webhook/webhook_client_impl.go

```diff
@@ -2,12 +2,36 @@ package webhook
 
 import (
 	"context"
+	"errors"
+	"net/url"
+	"strings"
 
 	"github.com/disgoorg/disgo/discord"
 	"github.com/disgoorg/disgo/rest"
 	"github.com/disgoorg/snowflake/v2"
 )
 
+// ParseURL parses a webhook URL
+func ParseURL(webhook string, opts ...ConfigOpt) (Client, error) {
+	u, err := url.Parse(webhook)
+	if err != nil {
+		return nil, err
+	}
+
+	parts := strings.FieldsFunc(u.Path, func(r rune) bool { return r == '/' })
```

Comment:
why do you not just use `strings.Split`?

Reply:
`Fields(Func)` will ignore otherwise empty parts, so `/a/b/c/` returns `[a b c]` (3 items) rather than `[ a b c ]` (5 items)
Which makes it easier to reason about than having to deal with ending slashes, etc.

Reply:
ah cool 👍🏻